core/src/main/java/com/linecorp/armeria/internal/common/util/IdentityHashStrategy.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright 2020 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.linecorp.armeria.internal.common.util;
+
+import it.unimi.dsi.fastutil.Hash.Strategy;
+
+public final class IdentityHashStrategy<T> implements Strategy<T> {
```

Comment:
Can we make this as an enum? 🤔

Reply:
Then we will lose the type parameter, i.e. the caller has to do an unchecked cast.

Reply:
Oops, I missed that. 😓